include/nektar_interface/particle_cell_mapping/particle_cell_mapping_2d.hpp

```diff
@@ -0,0 +1,137 @@
+
+  std::unique_ptr<ErrorPropagate> ep;
+
+  template <typename U>
+  inline void write_vertices_2d(U &geom, const int index, double *output) {
+    int last_point_index = -1;
+    if (geom->GetShapeType() == LibUtilities::eTriangle) {
+      last_point_index = 2;
+    } else if (geom->GetShapeType() == LibUtilities::eQuadrilateral) {
+      last_point_index = 3;
+    } else {
+      NESOASSERT(false, "get_local_coords_2d Unknown shape type.");
+    }
+    const auto v0 = geom->GetVertex(0);
+    const auto v1 = geom->GetVertex(1);
+    const auto v2 = geom->GetVertex(last_point_index);
+
+    NESOASSERT(v0->GetCoordim() == 2, "Expected v0->Coordim to be 2.");
+    NESOASSERT(v1->GetCoordim() == 2, "Expected v1->Coordim to be 2.");
+    NESOASSERT(v2->GetCoordim() == 2, "Expected v2->Coordim to be 2.");
+
+    output[index * 6 + 0] = (*v0)[0];
+    output[index * 6 + 1] = (*v0)[1];
+    output[index * 6 + 2] = (*v1)[0];
+    output[index * 6 + 3] = (*v1)[1];
+    output[index * 6 + 4] = (*v2)[0];
+    output[index * 6 + 5] = (*v2)[1];
+  }
+
+public:
+  /**
+   *  Create new instance for all 2D geometry objects in ParticleMeshInterface.
+   *
+   *  @param sycl_target SYCLTarget to use for computation.
+   *  @param particle_mesh_interface ParticleMeshInterface containing graph.
+   */
+  MapParticles2DRegular(SYCLTargetSharedPtr sycl_target,
+                        ParticleMeshInterfaceSharedPtr particle_mesh_interface);
+
+  /**
+   *  Called internally by NESO-Particles to map positions to Nektar++
+   *  triangles and quads.
+   */
+  void map(ParticleGroup &particle_group, const int map_cell = -1,
+           const double tol = 0.0);
+};
+
+/**
+ *  Class to map particles into Nektar++ cells.
+ */
+class MapParticles2D {
+protected:
+  /// Disable (implicit) copies.
+  MapParticles2D(const MapParticles2D &st) = delete;
+  /// Disable (implicit) copies.
+  MapParticles2D &operator=(MapParticles2D const &a) = delete;
+
+  SYCLTargetSharedPtr sycl_target;
+  ParticleMeshInterfaceSharedPtr particle_mesh_interface;
+
+  std::unique_ptr<MapParticlesCommon> map_particles_common;
+  std::unique_ptr<MapParticles2DRegular> map_particles_2d_regular;
+  std::unique_ptr<MapParticlesHost> map_particles_host;
+  std::unique_ptr<Newton::MapParticlesNewton<Newton::MappingQuadLinear2D>>
+      map_particles_newton_linear_quad;
+
+  int count_regular = 0;
+  int count_deformed = 0;
+
+public:
+  /**
+   *  Constructor for mapping class.
+   *
+   *  @param sycl_target SYCLTarget on which to perform mapping.
+   *  @param particle_mesh_interface ParticleMeshInterface containing 2D
+   * Nektar++ cells.
+   */
+  MapParticles2D(SYCLTargetSharedPtr sycl_target,
+                 ParticleMeshInterfaceSharedPtr particle_mesh_interface);
+
+  /**
+   *  Called internally by NESO-Particles to map positions to Nektar++
+   *  triangles and quads.
+   */
+  void map(ParticleGroup &particle_group, const int map_cell = -1,
```

Comment:
This tol is different to the one in the next file down. Is there a good reason? Perhaps pull out into a const variable in an anon namespace so they all share the same tol (if they indeed should?)?

Reply:
Added a generic key-value store `ParameterStore` and added that to the mapper constructors to set tolerances and iterations counts (with sensible defaults). This allows setting the parameters per mapping type.